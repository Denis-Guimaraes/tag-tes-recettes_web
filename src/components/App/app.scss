@import 'src/styles/vars';

.app {
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  font-size: $base-font-size;

  &-wrapper {
    flex-grow: 2;
  }
}
