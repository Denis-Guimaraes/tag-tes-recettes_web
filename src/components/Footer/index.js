// NPM import
import React from 'react';

// Local import

// Styles
import './footer.scss';

// Code
const getYear = () => {
  const date = new Date();
  return date.getFullYear();
};

const Footer = () => (
  <footer className="footer">
    <p className="footer__text">&copy; {getYear()} Denis Guimaraes.</p>
    <ul>
      <li className="footer__link">Mentions l√©gales</li>
      <li className="footer__link">Contact</li>
    </ul>
  </footer>
);

// Export
export default Footer;
