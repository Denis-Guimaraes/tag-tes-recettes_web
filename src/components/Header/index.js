// NPM import
import React from 'react';
import { Link } from 'react-router-dom';

// Local import

// Styles
import './header.scss';

// Code
const Header = () => (
  <header className="header">
    <h1 className="header__title">
      <Link to="/">
        Tag Tes Recettes
      </Link>
    </h1>
  </header>
);

// Export
export default Header;
