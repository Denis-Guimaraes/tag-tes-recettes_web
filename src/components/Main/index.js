// NPM import
import React from 'react';
import { Route } from 'react-router-dom';
import PropTypes from 'prop-types';

// Local import
import Home from 'src/components/Home';
import Signin from 'src/containers/Signin';
import Signup from 'src/containers/Signup';
import Modal from 'src/containers/Modal';
import Router from 'src/containers/Main/Router';

// Styles
import './main.scss';

// Code
const Main = ({ userMessage, userToken }) => {
  if (userToken.length > 0) {
    return (
      <Router />
    );
  }
  return (
    <main className="main">
      <Route exact path="/" component={Home} />
      <Route path="/connexion" component={Signin} />
      <Route path="/inscription" component={Signup} />
      {userMessage.length > 0 && <Modal />}
    </main>
  );
};

Main.propTypes = {
  userMessage: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  userToken: PropTypes.string.isRequired,
};

// Export
export default Main;
