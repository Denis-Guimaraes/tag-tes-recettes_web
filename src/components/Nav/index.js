// NPM import
import React from 'react';
import PropTypes from 'prop-types';

// Local import
import NavDisconnected from './NavDisconnected';
import NavConnected from './NavConnected';

// Styles
import './nav.scss';

// Code
const Nav = ({ userToken }) => (
  <nav className="nav">
    {userToken.length <= 0 ? <NavDisconnected /> : <NavConnected />}
  </nav>
);

Nav.propTypes = {
  userToken: PropTypes.string.isRequired,
};

// Export
export default Nav;
