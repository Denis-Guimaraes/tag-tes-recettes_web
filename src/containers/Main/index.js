// NPM import
import { connect } from 'react-redux';

// Local import
import Main from 'src/components/Main';

// Code
const mapStateToProps = state => ({
  userMessage: state.user.userMessage,
});

const mapDispatchToProps = () => ({});

// Export
export default connect(mapStateToProps, mapDispatchToProps)(Main);
