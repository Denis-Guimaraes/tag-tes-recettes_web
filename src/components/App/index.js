// NPM import
import React from 'react';

// Local import
import Header from 'src/components/Header';
import Nav from 'src/components/Nav';
import Main from 'src/containers/Main';
import Footer from 'src/components/Footer';

// Styles
import './app.scss';

// Code
const App = () => (
  <div className="app">
    <Header />
    <div className="app-wrapper">
      <Nav />
      <Main />
    </div>
    <Footer />
  </div>
);

// Export
export default App;
