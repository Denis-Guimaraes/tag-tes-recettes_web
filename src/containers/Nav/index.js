// NPM import
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

// Local import
import Nav from 'src/components/Nav';

// Code
const mapStateToProps = state => ({
  userToken: state.user.userToken,
});

const mapDispatchToProps = () => ({});

// Export
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Nav));
