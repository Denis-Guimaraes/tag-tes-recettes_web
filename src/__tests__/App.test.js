/* eslint-disable no-undef */
// NPM import
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
// Local import
import App from '../components/App';
import Header from '../components/Header';
import Nav from '../containers/Nav';
import Main from '../containers/Main';
import Footer from '../components/Footer';

describe('App', () => {
  it('match snapshot', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('contain component Header', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.exists(Header)).toBe(true);
  });
  it('contain component Nav', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.exists(Nav)).toBe(true);
  });
  it('contain component Main', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.exists(Main)).toBe(true);
  });
  it('contain Footer', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.exists(Footer)).toBe(true);
  });
});
