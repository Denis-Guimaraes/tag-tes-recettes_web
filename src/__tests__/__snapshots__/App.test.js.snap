// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App match snapshot 1`] = `
<div
  className="app"
>
  <Header />
  <div
    className="app-wrapper"
  >
    <withRouter(Connect(Nav)) />
    <withRouter(Connect(Main)) />
  </div>
  <Footer />
</div>
`;
